scripts/deploy_on_kind.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+set -e
+
+MR_NAMESPACE="${MR_NAMESPACE:-kubeflow}"
+
+# modularity to allow re-use this script against a remote k8s cluster
+if [[ -n "$LOCAL" ]]; then
+    CLUSTER_NAME="${CLUSTER_NAME:-kind}"
+    IMG="${IMG:-docker.io/kubeflow/model-registry:main}"
+
+    echo 'Creating local Kind cluster and loading image'
+
+    if [[ $(kind get clusters || false) =~ $CLUSTER_NAME ]]; then
+        echo 'Cluster already exists, skipping creation'
+
+        kubectl config use-context "kind-$CLUSTER_NAME"
+    else
+        kind create cluster -n "$CLUSTER_NAME"
+    fi
+
+    kind load docker-image -n "$CLUSTER_NAME" "$IMG"
+
+    echo 'Image loaded into kind cluster - use this command to port forward the mr service:'
+    echo "kubectl port-forward -n $MR_NAMESPACE service/model-registry-service 8080:8080 &"
+fi
+
+echo 'Deploying model registry to Kind cluster'
+if [[ $(kubectl get namespaces || false) =~ $MR_NAMESPACE ]]; then
+    echo 'Namespace already exists, skipping creation'
+else
+    kubectl create namespace "$MR_NAMESPACE"
+fi
+
+kubectl apply -k manifests/kustomize/overlays/db
+kubectl set image -n kubeflow deployment/model-registry-deployment rest-container="$IMG"
```

Comment:
to double-check.

during release, a PR with the updated manifest kustomization image setting is applied.
So first, the manifests are deployed to kind with an image tag which will not be found (the tag does not exists yet in the container registry).

This line (36) actually set the deployment model registry image to the one locally built, hence existing and loaded in the kind cluster.

Can this be confirmed, please?

Reply:
the image is being loaded on local mode inside the script. and on the CI it's loaded with a different command, but by the same method. So two cases here:
1. LOCAL envvar set:
  create the cluster and load the pre-built image inside, then run line 36 eventually.
2. No envvar set:
  assume the cluster already exists and has the image loaded. then we only set it after deploying MR (this avoids using kustomize directly which would change the manifests)
  
does this answer? otherwise @Al-Pragliola might be able to answer better as he suggested this change :)

Reply:
thanks @isinyaaa I see, in case of GHA we do not set the LOCAL and we invoke directly the pytest via nox (instead of make): 

https://github.com/kubeflow/model-registry/pull/447/files#diff-207f09dfa0bc604a684631a1c78be7d9aa00f9a6c778eec30afa287990be3da1R119-R143

that should do it. 
  
I will wait for confirmation from @Al-Pragliola and likely will add some comments into it.

(no need to reply over weekend 😅 but I wanted to mark it down before forgetting 🙃 thanks a lot)

Reply:
I confirm what @isinyaaa just said, the job of the deploy_on_kind script is to first fill all the requirements for deploying mr and then deploy it on the current's context cluster.

In case of CI all requirements:

- existing Kubernetes cluster
- mr docker image built
- mr docker image present on the cluster's registry

are already done in the previous steps of the action, so we can just create the namespace and deploy by not setting the LOCAL env.

We decided to replace kustomize edit with kubectl edit for two reasons:

1) remove a dependency
2) avoid creating drifts, since kustomize actually modifies the manifests, which is a bit of a pain when running the script locally because the user has to undo local changes. 

Hope this works well for everyone.